src/java/org/apache/cassandra/index/sai/StorageAttachedIndex.java

```diff
@@ -94,6 +94,8 @@
 
 public class StorageAttachedIndex implements Index
 {
+    public static final String NAME = "sai";
```

Comment:
what about add a new method named getIndexName() in the interface java Index.java , and every implementation index implement the method and return the real name ?

Reply:
But IndexMetada seems to use the name to build indexNameAliases ....

Reply:
See https://github.com/apache/cassandra/pull/2433#discussion_r1242403756

> We could always build a service provider framework or something here to register the index name aliases for custom 2i implementations and support them in CREATE INDEX...but not in this patch.

The other complication here is that `CassandraIndex` is an abstract class. It's subclasses were never meant to be exposed to the user, which makes the SPI thing a little more sticky.